unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs

```diff
@@ -6936,129 +6936,148 @@ public bool objectIsCurrentlyVisible(SimObjPhysics sop, float maxDistance)
                 return;
             }
-            Vector3[] positions = null;
-            if (action.positions != null && action.positions.Count != 0) {
-                positions = action.positions.ToArray();
-            } else {
-                positions = getReachablePositions();
-            }
+            SimObjPhysics theObject = physicsSceneManager.ObjectIdToSimObjPhysics[objectId];
 
+            // save current agent pose
             bool wasStanding = isStanding();
             Vector3 oldPosition = transform.position;
             Quaternion oldRotation = transform.rotation;
             Vector3 oldHorizon = m_Camera.transform.localEulerAngles;
+            if (ItemInHand != null) {
+                ItemInHand.gameObject.SetActive(false);
+            }
 
-            if (!physicsSceneManager.ObjectIdToSimObjPhysics.ContainsKey(action.objectId)) {
-                errorMessage = "Object ID appears to be invalid.";
-                actionFinished(false);
-                return;
+            // Populate default standings. Note that these are boolean because that's
+            // the most natural integration with Teleport
+            if (standings == null) {
+                standings = new bool[] {false, true};
             }
 
-            if (ItemInHand != null) {
-                ItemInHand.gameObject.SetActive(false);
+            // populate default horizons
+            if (horizons == null) {
+                horizons = new float[] {-30, 0, 30, 60};
             }
 
-            SimObjPhysics theObject = physicsSceneManager.ObjectIdToSimObjPhysics[action.objectId];
+            // populate the positions by those that are reachable
+            if (positions == null) {
+                positions = getReachablePositions();
+            }
+
+            // populate the rotations based on rotateStepDegrees
+            if (rotations == null) {
+                rotations = new float[(int) (360 / rotateStepDegrees)];
+                int i = 0;
+                for (float rotation = 0; rotation < 360; rotation += rotateStepDegrees) {
+                    rotations[i++] = rotation;
+                }
+            }
 
             // Don't want to consider all positions in the scene, just those from which the object
             // is plausibly visible. The following computes a "fudgeFactor" (radius of the object)
             // which is then used to filter the set of all reachable positions to just those plausible positions.
             Bounds objectBounds = new Bounds(
-                new Vector3(float.PositiveInfinity, float.PositiveInfinity, float.PositiveInfinity),
-                new Vector3(-float.PositiveInfinity, -float.PositiveInfinity, -float.PositiveInfinity)
+                center: new Vector3(float.PositiveInfinity, float.PositiveInfinity, float.PositiveInfinity),
+                size: new Vector3(-float.PositiveInfinity, -float.PositiveInfinity, -float.PositiveInfinity)
```

Comment:
With you writing out the parameters I'm suddenly surprised that this works. I know the intention is to have them be effectively undefined so that they become defined by the first call to `Encapsulate` but this does seem weird. Is it possible to have a small unit test that this behavior works correctly?

Reply:
Yeah, will try it. Also was kinda surprised that these were the parameters on bounds, haha.

Reply:
A test has been added (and passes)